File: CCharp Basic/04.ConsoleIO/06.QuadraticEquation/QuadraticEquation.cs
using System;
//Problem 6.	Quadratic Equation
//Write a program that reads the coefficients a, b and c 
//of a quadratic equation ax2 + bx + c = 0 and solves 
//it (prints its real roots). 

class QuadraticEquation
{
    static void Main()
    {
        double a = double.Parse(Console.ReadLine());
        double b = double.Parse(Console.ReadLine());
        double c = double.Parse(Console.ReadLine());

        // finding discriminant
        double d = (b * b) - (4 * a * c);

        if (d < 0 || a == 0.0)
        {
            Console.WriteLine("No real roots!");
        }
        else
        {
            Console.WriteLine("x1={0}; x2={1}",
                (-b - Math.Sqrt(d)) / (2 * a),
                (-b + Math.Sqrt(d)) / (2 * a));
        }
    }
}

File: JavaScript/02.JavaScript-Syntax/01.Quadratic-equation/quadratic-equation.js
/*
 * 01. Quadratic equation
 * Write a script that enters the coefficients a, b and c of a
 * quadratic equation a*x2 + b*x + c = 0 and calculates and prints
 * its real roots. Note that quadratic equations may have 0, 1 or 2 real roots.
 */

function getRealRoots(a, b, c) {
    var discriminant = Math.pow(b, 2) - 4 * a * c;
    if (discriminant > 0) {
        var x1 = (-b - Math.sqrt(discriminant)) / (2 * a),
            x2 = (-b + Math.sqrt(discriminant)) / (2 * a);
        console.log('x1 = ' + x1);
        console.log('x2 = ' + x2);
    } else if (discriminant === 0) {
        var x = -b / (2 * a);
        console.log('x = ' + x);
    } else {
        console.log('No real roots');
    }
}

getRealRoots(2, 5, -3);
getRealRoots(2, -4, 2);
getRealRoots(4, 2, 1);